### PR TITLE
Internal refactoring and documentation

### PR DESCRIPTION
**Pull request by zrzka**
* `.travis.yml` updated to build PRs & `master` branch only
* Forbid `unused_must_use` warning -> error
* Mark reexports in the documentation as reexports (`#[doc(no_inline)]`)
* Code refactoring in the same way as I did in the `-cursor` (& no breaking changes)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,8 +1,250 @@
...
+///
+/// - Clearing (all lines, current line, from cursor down and up, until new line)
+/// - Scrolling (Up, down)
+/// - Get the size of the terminal
+/// - Set the size of the terminal
+/// - Alternate screen
+/// - Raw screen
+/// - Exit the current process
+///
+/// Check `/examples/` in the library for more specific examples.
```
Comment (TimonPost) [inline src/lib.rs]: Let's add an example here

**Comment by zrzka** [comment]
@TimonPost removed `[WIP]` -> ready for review. Same stuff which was done for the `-cursor`.

**Comment by zrzka** [comment]
@TimonPost err, taking back, I forgot about the `README.md`.

**Comment by zrzka** [comment]
Fixed, ready.

File: README.md [hunk 2]
```diff
@@ -20,111 +20,52 @@ for more info).
...
-let (width, height) = terminal.size()?;
-print!("X: {}, y: {}", width, height);
-
-// Scroll down, up 10 lines.
-terminal.scroll_down(10)?;
-terminal.scroll_up(10)?;
+```toml
+[dependencies]
+# All crossterm features are enabled by default.
+crossterm = "0.11"
```
Comment (TimonPost) [inline README.md:45]: crossterm_input?

File: src/lib.rs [hunk 3]
```diff
@@ -1,8 +1,278 @@
...
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Ord, PartialOrd, Hash)]
+pub enum ClearType {
+    /// All cells.
+    All,
+    /// All cells from the cursor position downwards.
+    FromCursorDown,
+    /// All cells from the cursor position upwards.
+    FromCursorUp,
+    /// Current line.
```
Comment (TimonPost) [inline src/lib.rs]:
```suggestion
    /// All cells at the cursor row.
```

File: src/terminal/ansi.rs [hunk 4]
```diff
@@ -6,55 +6,49 @@ use crossterm_utils::{csi, write_cout, Result};
...
-                write_cout!(CLEAR_UNTIL_NEW_LINE)?;
-            }
+            ClearType::All => write_cout!(CLEAR_ALL_CSI_SEQUENCE)?,
+            ClearType::FromCursorDown => write_cout!(CLEAR_FROM_CURSOR_DOWN_CSI_SEQUENCE)?,
+            ClearType::FromCursorUp => write_cout!(CLEAR_FROM_CURSOR_UP_CSI_SEQUENCE)?,
+            ClearType::CurrentLine => write_cout!(CLEAR_FROM_CURRENT_LINE_CSI_SEQUENCE)?,
+            ClearType::UntilNewLine => write_cout!(CLEAR_UNTIL_NEW_LINE_CSI_SEQUENCE)?,
         };
+
+        if clear_type == ClearType::All {
```
Comment (TimonPost) [inline src/terminal/ansi.rs:48]: I am unsure if this is really needed here. Maybe we should re-validate this.

File: README.md (hunk 2 above)
Comment (zrzka) [reply README.md:45]: This is intentional. See the cursor's [Getting started](https://github.com/crossterm-rs/crossterm-cursor/blob/master/README.md#getting-started). Same here. Why intentional? Yep, it's a `crossterm_terminal` crate, but I kind of want to "force" users to use `crossterm` crate.

File: src/lib.rs (hunk 3 above)
Comment (zrzka) [reply src/lib.rs]: Fixed.

File: src/terminal/ansi.rs (hunk 4 above)
Comment (zrzka) [reply src/terminal/ansi.rs:48]: Then we should create an issue for this as this PR goal is no breaking / functional changes at all. Just internal refactor & docs.

File: src/lib.rs (hunk 1 above)
Comment (zrzka) [reply src/lib.rs]: Seems like an outdated comment?

**Review by TimonPost — APPROVED** [review]

**Pull request merged**